mtp_api/apps/security/managers.py

```diff
@@ -420,3 +434,16 @@ def _get_matching_rules(self, credit):
             if rule.applies_to(credit) and rule.triggered(credit):
                 matched_rule_codes.append(rule_code)
         return matched_rule_codes
+
+
+class CheckAutoAcceptRuleManager(models.Manager):
+
+    def get_active_auto_accept_for_credit(self, credit: Credit):
+        auto_accept_rule = self.filter(
+            debit_card_sender_details=credit.sender_profile.debit_card_details.first(),
```

Comment:
I don't know the models very well so apologies if it's a stupid question.

Is it possible that any of the related models in this chain `credit.sender_profile.debit_card_details` is not present? and you think we should guard/check against that? 

But maybe all of them will be present in the case of a credit paid via debit card so maybe this is fine, but worth asking ☺️

Reply:
it doesn't really make sense for a credit to have a check made without first having a sender profile and debit card details. i think this is good.
see `payment.models.create_security_check_if_needed_and_attach_profiles`